### PR TITLE
Store records in their own repository contexts

### PR DESCRIPTION
**Pull request by ledsoft**
Implements kbss-cvut/23ava-distribution#10

Note that these changes are not compatible with existing data.

### CODE REVIEW

**Comment by ledsoft** [comment]
Since this PR has not been merged, yet, the new changes implementing kbss-cvut/23ava-distribution#11 have been automatically added to it.

**Comment by ledsoft** [comment]
Also I noticed that there is a [draft](https://github.com/orgs/kbss-cvut/projects/10/views/1#) of a ticket for migrating Record Manager from RDF4J to GraphDB. The Docker Compose setup I created uses GraphDB, so maybe that ticket can be closed as well.

File: docker-compose.yml [hunk 1]
```diff
@@ -1,53 +1,102 @@
...
+      NAVIGATOR_LANGUAGE: "true"
+      API_URL: "http://localhost:8080/record-manager"
+      APP_INFO: "<a href=\"https://github.com/kbss-cvut\" target=\"_blank\" rel=\"noopener noreferrer\" title=\"KBSS CVUT v Praze\">© KBSS CVUT v Praze, 2023</a>"
+      AUTHENTICATION: "oidc"
+      AUTH_SERVER_URL: "http://localhost:8088/realms/record-manager"
+      AUTH_CLIENT_ID: "record-manager-ui"
+      FORCE_BASENAME: "true"
 
-  dm-record-manager-server:
+  record-manager:
```
Comment (blcham) [inline docker-compose.yml:22]:
```suggestion
  record-manager-server:
```

File: docker-compose.yml [hunk 2]
```diff
@@ -1,53 +1,102 @@
...
     build: .
-    image: record-manager-server
-    container_name: dm-record-manager-server
+    image: record-manager
```
Comment (blcham) [inline docker-compose.yml:24]:
```suggestion
    image: record-manager-server
```

File: docker-compose.yml [hunk 3]
```diff
@@ -1,53 +1,102 @@
 version: '3.9'
 
 services:
-  dm-record-manager:
-    image: 'ofn-record-manager:latest'
-    container_name: dm-record-manager
+  record-manager-ui:
```
Comment (blcham) [inline docker-compose.yml:4]:
```suggestion
  record-manager:
```

File: db-server/repo-config/config-record-manager.ttl [hunk 4]
```diff
@@ -0,0 +1,33 @@
+@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
+@prefix rep: <http://www.openrdf.org/config/repository#> .
+@prefix sail: <http://www.openrdf.org/config/sail#> .
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
+@prefix graphdb: <http://www.ontotext.com/config/graphdb#>.
+
+<#termit> a rep:Repository;
```
Comment (blcham) [inline db-server/repo-config/config-record-manager.ttl:7]:
```suggestion
<#record-manager-app> a rep:Repository;
```

File: docker-compose.yml [hunk 5]
```diff
@@ -1,53 +1,102 @@
...
+      AUTH_CLIENT_ID: "record-manager-ui"
+      FORCE_BASENAME: "true"
 
-  dm-record-manager-server:
+  record-manager:
     build: .
-    image: record-manager-server
-    container_name: dm-record-manager-server
+    image: record-manager
+    container_name: record-manager
```
Comment (blcham) [inline docker-compose.yml:25]:
```suggestion
    container_name: record-manager-server
```

File: docker-compose.yml [hunk 6]
```diff
@@ -1,53 +1,102 @@
...
-      - dm-s-pipes-engine
-      - dm-rdf4j
+      - s-pipes-engine
+      - db-server
+      - auth-server
     environment:
-      repositoryUrl: "http://dm-rdf4j:8080/rdf4j-server/repositories/ofn-form-manager-app"
-      formGenRepositoryUrl: "http://dm-rdf4j:8080/rdf4j-server/repositories/ofn-form-manager-formgen"
-      formGenServiceUrl: "http://dm-s-pipes-engine:8080/s-pipes/service?_pId=clone&sgovRepositoryUrl=https%3A%2F%2Fgraphdb.onto.fel.cvut.cz%2Frepositories%2Fkodi-slovnik-gov-cz"
+      REPOSITORYURL: "http://db-server:7200/repositories/record-manager"
```
Comment (blcham) [inline docker-compose.yml:33]:
```suggestion
      REPOSITORYURL: "http://db-server:7200/repositories/record-manager-app"
```

File: docker-compose.yml [hunk 7]
```diff
@@ -1,53 +1,102 @@
...
-    image: 'ofn-record-manager:latest'
-    container_name: dm-record-manager
+  record-manager-ui:
+    image: 'ghcr.io/kbss-cvut/kbss-cvut/record-manager-ui:latest'
     ports:
-      - '4000:80'
+      - "127.0.0.1:3000:80"
     depends_on:
-      - dm-record-manager-server
+      - record-manager
```
Comment (blcham) [inline docker-compose.yml:9]:
```suggestion
      - record-manager-server
```

File: docker-compose.yml [hunk 8]
```diff
@@ -1,53 +1,102 @@
...
     environment:
-      API_URL: "http://localhost:3000/ofn-record-manager"
+      APP_TITLE: "Record Manager"
+      BASENAME: "./"
+      LANGUAGE: "cs"
+      NAVIGATOR_LANGUAGE: "true"
+      API_URL: "http://localhost:8080/record-manager"
```
Comment (blcham) [inline docker-compose.yml:15]:
```suggestion
      API_URL: "http://localhost:8080/record-manager-server"
```

Comment (blcham) [reply docker-compose.yml:15]: Maybe this requires also "context-path" setting in  service "record-manager-server".

**Review by blcham — CHANGES_REQUESTED** [review]
LGTM, just minor suggestions.

**Pull request merged**